lib/Core/Executor.cpp

```diff
@@ -958,40 +966,61 @@ void Executor::branch(ExecutionState &state,
       addConstraint(*result[i], conditions[i]);
 }
 
+ref<Expr> Executor::maxStaticPctChecks(ExecutionState &current,
+                                       ref<Expr> condition) {
+  if (isa<klee::ConstantExpr>(condition))
+    return condition;
+
+  if (MaxStaticForkPct == 1. && MaxStaticSolvePct == 1. &&
+      MaxStaticCPForkPct == 1. && MaxStaticCPSolvePct == 1.)
+    return condition;
+
+  // these checks are performed only after MaxStaticPctCheckDelay time has
+  // passed since execution started
+  if (statsTracker->elapsed() <= maxStaticPctCheckDelay)
+    return condition;
+
+  StatisticManager &sm = *theStatisticManager;
+  CallPathNode *cpn = current.stack.back().callPathNode;
+  if ((MaxStaticForkPct < 1. &&
```

Comment:
Can you divide this into subsequent separate boolean expressions:
```
bool reached_max_fork_limit = (sm.getIndexedValue(stats::forks, sm.getIndex()) >
           stats::forks * MaxStaticForkPct);
bool reached_max_solver_limit = (...)

if (!reached_max_fork_limit && !reached_max_solver_limit ...)
  return condition;

//concretize expression

```
This would simplify the understanding process and makes things more clear.

This could be also used to print the actual reason as part of the conretization.